packages/frontend/src/hooks/useSelectedNetwork.ts

```diff
@@ -0,0 +1,79 @@
+import { useQuery } from 'react-query'
+import { Token } from '@hop-protocol/sdk'
+import Network from 'src/models/Network'
+import { Addressish } from 'src/models/Address'
+import { useQueryParams } from '.'
+import { ChangeEvent, useEffect, useState } from 'react'
+import { defaultL2Network, l2Networks } from 'src/config/networks'
+import { findNetworkBySlug } from 'src/utils'
+
+async function fetchBalance(token: Token, address: Addressish) {
+  return token.balanceOf(address?.toString())
+}
+
+interface Options {
+  l2Only: boolean
+}
+
+export function useSelectedNetwork(opts: Options = { l2Only: false }) {
+  const [selectedNetwork, setSelectedNetwork] = useState<Network>(defaultL2Network)
+  const { queryParams, updateQueryParams } = useQueryParams()
+
+  useEffect(() => {
+    if (queryParams?.sourceNetwork !== selectedNetwork.slug) {
+      const matchingNetwork = findNetworkBySlug(queryParams.sourceNetwork)
```

Comment:
I get this type error on start

![DeepinScreenshot_select-area_20220125172921](https://user-images.githubusercontent.com/168240/151087897-4ed71297-f151-4aa3-bb13-327b2df36e36.png)

Type asserting as string solves the issue

 `queryParams.sourceNetwork as string`